api/turing/config/config.go

```diff
@@ -100,7 +107,58 @@ type DeploymentConfig struct {
 
 // EnsemblingJobConfig captures the config related to the ensembling batch jobs.
 type EnsemblingJobConfig struct {
-	DefaultEnvironment string `validate:"required"`
+	BatchSize          int          `validate:"required"`
+	MaxRetryCount      int          `validate:"required"`
+	DefaultEnvironment string       `validate:"required"`
+	KanikoConfig       KanikoConfig `validate:"required"`
+	ImageConfig        ImageConfig  `validate:"required"`
+	InjectGojekConfig  bool
```

Comment:
`InjectGojekConfig` feels adhoc. I'm wondering if there is a better way to control this, perhaps with a label map in the API config that should be applied to the resources deployed ? Whatever mechanism we choose should probably also be adopted for the labels applied to the router, enricher, etc. ([ref](https://github.com/gojek/turing/blob/main/api/turing/cluster/servicebuilder/service_builder.go#L331)).

Reply:
I'm also not very sure on what the best way to approach this as well @romanwozniak also brought this up here. https://github.com/gojek/turing/pull/61#discussion_r646595471

The ref you showed basically forces people to use our Gojek internal labels, but I thought that making a switch here would be simpler, since there is some logic involved in how the fields should be populated. I'm open to suggestions here since I'm also stuck at this.